app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorEndToEnd.kt

```diff
@@ -0,0 +1,251 @@
+    composeTestRule.onNodeWithTag("roleButtonTutor").performClick()
+    composeTestRule.onNodeWithTag("sectionDropdown").performClick()
+    composeTestRule.onNodeWithTag("sectionDropdownItem-SC").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdown").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdownItem-BA3").performClick()
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+    assertEquals(Role.TUTOR, mockProfileViewModel.currentProfile.value?.role)
+
+    // Create Tutor Profile Screen
+    composeTestRule.onNodeWithTag("welcomeText").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("checkbox_FRENCH").performClick()
+    composeTestRule.onNodeWithTag("checkbox_ENGLISH").performClick()
+    composeTestRule.onNodeWithTag("subjectButton").performClick()
+    composeTestRule.onNodeWithTag("dropdownANALYSIS").performClick()
+    composeTestRule.onNodeWithTag("dropdownPHYSICS").performClick()
+    composeTestRule.onNodeWithTag("subjectButton").performClick()
+    composeTestRule.onNodeWithTag("priceSlider").performGesture { swipeRight() }
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+
+    // Create Tutor Schedule Screen
+    composeTestRule
+        .onNodeWithTag("welcomeText")
+        .assertTextEquals("John, show us your availabilities")
+    composeTestRule.onNodeWithTag("Slot_0_0").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_3").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_2").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_6").performClick()
+    composeTestRule.onNodeWithTag("FindStudentButton").performClick()
+
+    // Home Screen
+    composeTestRule.onNodeWithContentDescription("Profile Icon").performClick()
+    composeTestRule.onNodeWithTag("profileStatus").assertTextEquals("Status: BA3 Tutor")
+    composeTestRule.onNodeWithTag("profileSection").assertTextEquals("Section: SC")
+    composeTestRule.onNodeWithTag("profilePrice").assertTextEquals("Price: 50.- per hour")
+    composeTestRule
+        .onNodeWithTag("lessonsCount")
+        .assertTextEquals("0 lessons given since you joined PocketTutor")
+    composeTestRule.onNodeWithTag("closeButton").performClick()
+    composeTestRule.onNodeWithContentDescription("Profile Icon").assertExists()
+    composeTestRule.onNodeWithTag("Find a Student").performClick()
+    composeTestRule.onNodeWithTag("screenTitle").assertExists()
+    composeTestRule.onNodeWithText("Physics Tutoring").assertIsDisplayed()
+    composeTestRule.onNodeWithText("physics").assertIsDisplayed()
+    composeTestRule.onNodeWithText("physics").performClick()
+    Thread.sleep(5000)
+    composeTestRule.onNodeWithTag("tutorLessonResponseScreen").assertExists()
+    composeTestRule.onNodeWithText("Ozymandias Halifax").assertExists()
+    composeTestRule.onNodeWithText("Offer to Teach (50.-/hour)").assertExists()
```

Comment:
These assertions are quite nice for completeness's sake. Since unit tests already check most of these smaller UI details, so many of them might not be necessary, but it is better to be too thorough than not enough!